app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, :all
+    return unless user
+
+    can %i[read create destroy], [Post, Comment, Like]
+    # can :read, :create, :destroy, Post
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
```

Comment:
- [ ] it would be better to fix your code so only the author of the post should delete it unless the admin can delete any right now I'm able to delete someone else post so please make sure to fix and also the same thing with  `comments`

Reply:
@youmari ok I will try to fix this but I need help. Please call me on Zoom.